apache-rat-core/src/main/java/org/apache/rat/report/xml/writer/impl/base/XmlWriter.java

```diff
@@ -514,6 +526,17 @@ private void writeAttributeContent(CharSequence content) throws IOException {
         writeEscaped(content, true);
     }
 
+    private void prepareForData() throws IOException {
+        if (elementNames.isEmpty()) {
+            if (elementsWritten) {
+                throw new OperationNotAllowedException("Root element has already been closed.");
+            } 
+            throw new OperationNotAllowedException("An element must be opened before content can be written.");
+        }
+        if (inElement) {
+            writer.write('>');
+        }
+    }
     /**
      * Writes content. Calling this method will automatically Note that this method
```

Comment:
Calling this method will automatically .... seems like a copy&paste thing - 
Do you want to add info that the element is closed automtically?